From e36aa0073d8620b09fd0ff5a335c5ea21d23d32b Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Sat, 2 Mar 2024 17:56:21 +0100
Subject: [PATCH] [knewfilemenu] Fix error handling when creating a folder that
 already exists

And setSelectDirWhenAlreadyExist is not set

In that case we want to do the error handling (and not assert)

BUG: 482145
---
 src/filewidgets/knewfilemenu.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/filewidgets/knewfilemenu.cpp b/src/filewidgets/knewfilemenu.cpp
index f9a630a5d9..cf4bab57e8 100644
--- a/src/filewidgets/knewfilemenu.cpp
+++ b/src/filewidgets/knewfilemenu.cpp
@@ -1494,10 +1494,9 @@ void KNewFileMenu::setSupportedMimeTypes(const QStringList &mime)
 void KNewFileMenu::slotResult(KJob *job)
 {
     if (job->error()) {
-        if (job->error() == KIO::ERR_DIR_ALREADY_EXIST) {
+        if (job->error() == KIO::ERR_DIR_ALREADY_EXIST && d->m_selectDirWhenAlreadyExists) {
             auto *simpleJob = ::qobject_cast<KIO::SimpleJob *>(job);
             if (simpleJob) {
-                Q_ASSERT(d->m_selectDirWhenAlreadyExists);
                 const QUrl jobUrl = simpleJob->url();
                 // Select the existing dir
                 Q_EMIT selectExistingDir(jobUrl);
-- 
GitLab

